apps/backend/src/iam/services/authentication/authentication.service.ts

```diff
@@ -0,0 +1,145 @@
+import { UserInputError } from '@nestjs/apollo';
+import { ExecutionContext, Inject, Injectable } from '@nestjs/common';
+import { ConfigType } from '@nestjs/config';
+import { Context, GqlExecutionContext } from '@nestjs/graphql';
+import { JwtService } from '@nestjs/jwt';
+import { randomUUID } from 'crypto';
+import { Request, Response } from 'express';
+import jwtConfig from 'src/iam/config/jwt.config';
+import { SignInInput } from 'src/iam/dto/sign-in.input';
+import { SignUpInput } from 'src/iam/dto/sign-up.input';
+import {
+  COOKIES_ACCESS_TOKEN_KEY,
+  COOKIES_REFRESH_TOKEN_KEY,
+} from 'src/iam/iam.constants';
+import { ActiveUserData } from 'src/iam/interfaces/active-user-data.interface';
+import { RefreshTokenIdsStorage } from 'src/iam/storage/refresh-token-ids.storage/refresh-token-ids.storage';
+import { User } from 'src/users/entities/user.entity';
+import { UsersService } from 'src/users/services/users/users.service';
+import { HashingService } from '../hashing/hashing.service';
+
+@Injectable()
+export class AuthenticationService {
+  @Inject(HashingService)
+  private readonly hashingService!: HashingService;
+  @Inject(UsersService)
+  private readonly usersService!: UsersService;
+  @Inject(JwtService)
+  private readonly jwtService!: JwtService;
+  @Inject(jwtConfig.KEY)
+  private readonly jwtConfiguration!: ConfigType<typeof jwtConfig>;
+  @Inject(RefreshTokenIdsStorage)
+  private readonly refreshTokenIdsStorage!: RefreshTokenIdsStorage;
+
+  async signUp(signUpInput: SignUpInput): Promise<User> {
+    const encryptedPassword = await this.hashingService.hash(
+      signUpInput.password,
+    );
+    return this.usersService.create({
+      ...signUpInput,
+      encryptedPassword,
+    });
+  }
+
+  async signIn(signInInput: SignInInput) {
+    const user = await this.usersService.findByEmail(signInInput.email);
```

Comment:
the user service should return null if the user does not exist because sometimes you don't want to throw an error and do a condition but the function will return an error